### DISCUSSION TITLE
Seeking GUI Frontend for PowerShell VLAN Commands

### DISCUSSION THREAD

**Discussion by luckygreen** [ChrisTitusTech/winutil#666] [Q&A]
Greetings! I am looking for a pointer to a pointer. In which project or chat server might one find folks that slap GUIs onto Windows PowerShell commands?

I am looking for a GUI-based solution for configuring VLANs on Windows 11 NICs since the NIC manufacturers GUI tools, [notably IntelPROSet, do not support Windows 11](https://www.intel.com/content/www/us/en/download/727998/intel-network-adapter-driver-for-microsoft-windows-11.html), requiring the user to employ contortions, such as the below, to configure VLANs on NICs on bare metal Windows 11 hosts:

```
Get-NetAdapter
New-VMSwitch -name VLAN-vSwitch -NetAdapterName Ethernet -AllowManagementOS $true
Remove-VMNetworkAdapter -ManagementOS -Name VLAN-vSwitch
Add-VMNetworkAdapter -ManagementOS -Name "VLAN123" -SwitchName "VLAN-vSwitch" -Passthru | Set-VMNetworkAdapterVlan -Access -VlanId 123
Add-VMNetworkAdapter -ManagementOS -Name "VLAN456" -SwitchName "VLAN-vSwitch" -Passthru | Set-VMNetworkAdapterVlan -Access -VlanId 456

```
Yes, I looked on GitHub. Nothing here that I can find. I was able to find a [single website discussing the underlying PowerShell commands](https://taeffner.net/2022/04/multiple-vlans-windows-10-11-onboard-tools-hyper-v/), but have not been able to find a GUI version.

Suggestions where to best ask this question would be much appreciated!

**Comment by cabca** [comment]
You can ask ChatGPT.


`Add-Type -AssemblyName System.Windows.Forms
$form = New-Object System.Windows.Forms.Form
$form.Text = "Configure VLAN"
$form.Width = 300
$form.Height = 200
$form.StartPosition = "CenterScreen"

$lblVlanId = New-Object System.Windows.Forms.Label
$lblVlanId.Location = New-Object System.Drawing.Point(10,20)
$lblVlanId.Size = New-Object System.Drawing.Size(120,20)
$lblVlanId.Text = "VLAN ID:"
$form.Controls.Add($lblVlanId)

$txtVlanId = New-Object System.Windows.Forms.TextBox
$txtVlanId.Location = New-Object System.Drawing.Point(130,20)
$txtVlanId.Size = New-Object System.Drawing.Size(150,20)
$form.Controls.Add($txtVlanId)

$lblNicName = New-Object System.Windows.Forms.Label
$lblNicName.Location = New-Object System.Drawing.Point(10,50)
$lblNicName.Size = New-Object System.Drawing.Size(120,20)
$lblNicName.Text = "NIC name:"
$form.Controls.Add($lblNicName)

$txtNicName = New-Object System.Windows.Forms.TextBox
$txtNicName.Location = New-Object System.Drawing.Point(130,50)
$txtNicName.Size = New-Object System.Drawing.Size(150,20)
$form.Controls.Add($txtNicName)

$btnExecute = New-Object System.Windows.Forms.Button
$btnExecute.Location = New-Object System.Drawing.Point(100,100)
$btnExecute.Size = New-Object System.Drawing.Size(100,30)
$btnExecute.Text = "Execute"
$form.Controls.Add($btnExecute)

$btnExecute.Add_Click({
    $vlanId = $txtVlanId.Text
    $nicName = $txtNicName.Text
    $vSwitchName = "VLAN-vSwitch"

    # Execute the PowerShell command to configure VLANs
    New-VMSwitch -name $vSwitchName -NetAdapterName $nicName -AllowManagementOS $true
    Remove-VMNetworkAdapter -ManagementOS -Name $vSwitchName
    Add-VMNetworkAdapter -ManagementOS -Name "VLAN$vlanId" -SwitchName $vSwitchName -Passthru | Set-VMNetworkAdapterVlan -Access -VlanId $vlanId

    [System.Windows.Forms.MessageBox]::Show("VLAN $vlanId configured on $nicName.", "Success")
})

$form.ShowDialog() | Out-Null
`

or

`using System.Management.Automation;
using System.Management.Automation.Runspaces;

private void button1_Click(object sender, EventArgs e)
{
    // Get the values of the VLAN ID and NIC name textboxes
    string vlanId = textBox1.Text;
    string nicName = textBox2.Text;

    // Create a PowerShell command to configure VLANs
    string cmd = "New-VMSwitch -Name VLAN-vSwitch -NetAdapterName " + nicName + " -AllowManagementOS $true;" +
                 "Remove-VMNetworkAdapter -ManagementOS -Name VLAN-vSwitch;" +
                 "Add-VMNetworkAdapter -ManagementOS -Name VLAN" + vlanId + " -SwitchName VLAN-vSwitch -Passthru | Set-VMNetworkAdapterVlan -Access -VlanId " + vlanId;
    PowerShell ps = PowerShell.Create();
    ps.AddScript(cmd);

    // Execute the PowerShell command
    Runspace runspace = RunspaceFactory.CreateRunspace();
    runspace.Open();
    ps.Runspace = runspace;
    ps.Invoke();

    // Display a message box to indicate that the VLAN has been configured
    MessageBox.Show("VLAN " + vlanId + " has been configured on NIC " + nicName, "VLAN Configuration", MessageBoxButtons.OK, MessageBoxIcon.Information);
}
`
